src/dir/Dir.re

```diff
@@ -106,55 +107,73 @@ module WinConst = {
     };
 };
 
+/*
+ * The [Fs] API expects all slashes to be normalized to forward slashes,
+ * so we need to make sure we match that constraint for Windows.
+ */
+let normalizePathSeparator = {
+  let backSlashRegex = Str.regexp("\\\\");
+  pathStr => pathStr |> Str.global_replace(backSlashRegex, "/");
+};
+
+let normalizeIfWindows = pathStr => {
+  Sys.win32 ? normalizePathSeparator(pathStr) : pathStr;
+};
+
 /*
  * Might want to rethink the ones that throw on invalid absolute paths in case
  * an application wants to let the app startup with a bad environment, then fix
  * it, and then reload all the env here.
  */
-let getOptionalEnvAbsoluteExn = s =>
+let getOptionalEnvAbsoluteExn = s => {
   switch (Sys.getenv(s)) {
   | exception Not_found => None
-  | txt => Some(Fp.absoluteExn(txt))
+  | txt => txt |> normalizeIfWindows |> Fp.absoluteExn |> (v => Some(v))
   };
+};
+
 let getEnvAbsoluteExn = s =>
   switch (Sys.getenv(s)) {
   | exception Not_found =>
     raise(
       Invalid_argument("Environment variable " ++ s ++ " does not exist."),
     )
-  | txt => Fp.absoluteExn(txt)
+  | txt => txt |> normalizeIfWindows |> Fp.absoluteExn
   };
 
+let getOrThrow =
+  fun
+  | Some(v) => v
+  | None => raise(Invalid_argument("Expected some"));
```

Comment:
I imagine you might want a more descriptive message so that user reports are more clear. `getOrThrow(~msg, x)`
```reason
let getOrThrow =
  fun
  | Some(v) => v
  | None => raise(Invalid_argument("Expected some"));
```

Reply:
Good call! I used the friendly name from the mock env var to produce something a bit more understandable.